CHANGELOG.md

```diff
@@ -8,6 +8,16 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Changed
+
+- Redesign `go.opentelemetry.io/otel/log/logtest`. (#6342)
+  - Change `Recorder.Result` to return `Recording`.
+  - Add `Recording`, `Scope`, `Record` types.
+  - `Recorder` no longer separately stores records emitted by loggers with the same instrumentation scope.
+  - Add a testable example showing how `go.opentelemetry.io/otel/log/logtest` can be used.
+  - Remove `ScopeRecords`, `EmittedRecord`, `RecordFactory` types.
+  - Remove `AssertRecordEqual` function.
```

Comment:
These need to be moved to the `Removed` section of the changelog.

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/6342/commits/34625462218b9c5e680f46fa4e1bb1677c4377c0